packages/custom_lint/lib/src/package_utils.dart

```diff
@@ -78,7 +84,9 @@ Future<Pubspec?> tryParsePubspec(Directory directory) async {
 /// Throws if the parsing fails, such as if the file is badly formatted or
 /// does not exists.
 Future<Pubspec> parsePubspec(Directory directory) async {
-  return Pubspec.parse(await directory.pubspec.readAsString());
+  final pubspec =
+      findProjectDirectory(directory, missingPackageConfigOkay: true).pubspec;
```

Comment:
It shouldn't be the responsibility of the "parse" functions to "find" the project. 

The directory passed should already be a valid project directory.

Reply:
Done